components/query/feature-popup.component.js

```diff
@@ -34,19 +38,38 @@ export default {
           return feature.get('label');
         }
       },
+
       isFeatureRemovable(feature) {
         return HsQueryVectorService.isFeatureRemovable(feature);
       },
+
       isLayerEditable(layer) {
         return HsQueryVectorService.isLayerEditable(layer);
       },
-      removeFeature(feature) {
-        HsQueryVectorService.removeFeature(feature);
+
+      async removeFeature(feature) {
+        const confirmed = await HsConfirmDialogService.show(
+          gettext('Really delete all the feature?'),
+          gettext('Confirm delete')
+        );
+        if (confirmed == 'yes') {
+          HsQueryVectorService.removeFeature(feature);
+          HsQueryBaseService.featuresUnderMouse = [];
+        }
       },
-      clearLayer(layer) {
-        layer.getSource().clear();
+
+      async clearLayer(layer) {
+        const confirmed = await HsConfirmDialogService.show(
+          gettext('Really delete all the features in layer?'),
```

Comment:
I would change this to

`'Really delete all features from this layer?'`

or with template literals it could be

`'Really delete all features from layer "${layer.get('title'}"?'`

Reply:
Templite literals create a bit of problems with gettext which I dont know how to solve exactly right now.  
`gettext('Really delete all features from layer "{0}"?').replace('{0}', layer.get('title'), ` can work

Reply:
Hi, I had similar problem, possible solution (complete gettextCatalog required) is (adapt to your usecase :) ):
gettextCatalog.getString(
            'Experiment {{name}} model is being computed',
            { name: experiment.name },
          ),